src/Compilers/Core/Portable/MetadataReference/ModuleMetadata.cs

```diff
@@ -85,14 +96,39 @@ public static ModuleMetadata CreateFromMetadata(IntPtr metadata, int size)
                 throw new ArgumentOutOfRangeException(CodeAnalysisResources.SizeHasToBePositive, nameof(size));
             }
 
-            return new ModuleMetadata(metadata, size, includeEmbeddedInteropTypes: false, ignoreAssemblyRefs: false);
+            return new ModuleMetadata(metadata, size, includeEmbeddedInteropTypes: false, ignoreAssemblyRefs: false, owner: null, disposeOwner: false);
+        }
+
+        /// <summary>
+        /// Create metadata module from a raw memory pointer to metadata directory of a PE image or .cormeta section of an object file.
+        /// Only manifest modules are currently supported.
+        /// </summary>
+        /// <param name="stream">Stream containing raw memory.</param>
+        /// <param name="leaveOpen">
+        /// False to close the stream upon disposal of the metadata (the responsibility for disposal of the stream is
+        /// transferred upon entry of the constructor unless the arguments given are invalid).
+        /// </param>
+#pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads
+        public static unsafe ModuleMetadata CreateFromMetadata(UnmanagedMemoryStream stream, bool leaveOpen = false)
```

Comment:
>UnmanagedMemoryStream [](http://example.com/codeflow?start=63&length=21)

I think it might be better to add `(IntPtr metadata, int size, IDisposable owner, bool disposeOwner)` instead of UnmanagedMemoryStream specific API.

Reply:
sgtm.

Reply:
wfm.

Reply:
done.